FSE: Attempt adding SPT tests to CI conf

getdave opened the pull request:
#### Changes proposed in this Pull Request

As per `paAmJe-P4-p2` this attempts to add automated tests runs for the FSE Plugin to the Circle CI config.

getdave committed:
Attempt adding SPT tests to CI conf

getdave committed:
Add setup requirement.

getdave committed:
Fix invalid `npm install` command

simison commented on .circleci/config.yml:
```diff
@@ -323,7 +323,14 @@ jobs:
+    working_directory: ~/wp-calypso/apps/full-site-editing
+    steps:
+      - prepare
+      - run: npm install
```
`npm ci` would be little faster and fail if the lockfile is out of sync.

https://docs.npmjs.com/cli/ci.html

getdave replied:
Thanks. I wasn't aware of this. I've updated it.

simison commented on .circleci/config.yml:
```diff
@@ -323,7 +323,14 @@ jobs:
+    steps:
+      - prepare
+      - run: npm install
+      - run: npm run test:unit
```
Other sections above have `name` defined. Should this one too?

getdave replied:
Good point. Updated.

simison reviewed (commented):
Looking pretty good!
I think you can test this by taking out the draft state and Circle I will run the config. Edit: actually it does look like it's running also as draft?

getdave committed:
Update to name config command steps

getdave committed:
Update to use preferred npm ci to install

getdave committed:
Updates test snapshots to ensure all tests pass

Snapshots had not been updated by SPT team as automated test validation was not in place until this PR lands.

getdave commented:
Ok I now have the FSE tests running (and passing!) as part of the CI setup.

Waiting on further reviews before merging.

getdave committed:
Fix broken indentation syntax in config

getdave committed:
Fix the config syntax…again

getdave commented:
> @automattic/calypso-color-schemes not accessible from @automattic/calypso-build

For some reason I'm now getting a failure with this error.

simison commented:
I think when your run tests locally, you have root deps installed.

In circle ci you don't I suppose? Not sure what's a good way to go around that if that's the case.

sirreal commented:
I looked at the current errors in the build, I suspect the problem is that this package doesn't operate as an isolated package, but also isn't an integrated monorepo package.

Based on previous conversations, it's closer to an isolated package, however this line violates that:

https://github.com/Automattic/wp-calypso/blob/6c324d37c52b716ecc240d5f48612c1397512d98/apps/full-site-editing/package.json#L38

This is effectively depending on the "live" calypso-build in the monorepo. It may not work well when depended on by this semi-isolated package.

I'd suggest `npm install -D @automattic/calypso-build` on the FSE package as the shortest way to get this working as expected.

getdave commented:
Thanks for your advice @sirreal.

> I'd suggest npm install -D @automattic/calypso-build on the FSE package as the shortest way to get this working as expected.

@gwwar How would you feel about this?

getdave committed:
Update to add calypso-build as a direct devDep

Addresses https://github.com/Automattic/wp-calypso/pull/36556#issuecomment-539174287

gwwar commented:
> How would you feel about adding @automattic/calypso-build as a explicit dep.

Go for it 👍 I prefer explicit dependencies unless we don't have the time to actually publish a package, or if intertwined dependencies need to be strictly in sync.

getdave commented:
Looking for an approving 👍 here...

simison reviewed (approved):
Great to have tests run in CI, well done!

We talked a bit about how in-monorepo-but-not-part-of-monorepo situation of FSE might be causing funky dependency deduping issues, especially with `calypso-build` dependency. (p1570560903135700-slack-team-calypso) This might need revisiting later on but in my understanding this is totally fine as of now and safe to merge. 👍
